src/ui/widgets/ProgressBar/progressBar.tsx

```diff
@@ -22,65 +22,41 @@ export const ProgressBarProps = {
   horizontal: BoolPropOpt,
   showLabel: BoolPropOpt,
   fillColor: ColorPropOpt,
+  backgroundColor: ColorPropOpt,
   precision: IntPropOpt,
   font: FontPropOpt,
-  border: BorderPropOpt
+  border: BorderPropOpt,
+  height: FloatPropOpt,
+  width: FloatPropOpt
 };
 
 export const ProgressBarComponent = (
   props: InferWidgetProps<typeof ProgressBarProps> & PVComponent
 ): JSX.Element => {
-  const style = commonCss(props);
   const {
     value,
     limitsFromPv = false,
     showLabel = false,
     font,
     horizontal = true,
-    fillColor = "#00aa00",
+    fillColor = "#3CFF3C",
+    backgroundColor = "#FAFAFA",
     precision = undefined,
-    logScale = false
+    logScale = false,
+    width = WIDGET_DEFAULT_SIZES["progressbar"][0],
+    height = WIDGET_DEFAULT_SIZES["progressbar"][1]
   } = props;
-  let { min = 1e-10, max = 100 } = props;
 
+  let { min = 0, max = 100 } = props;
   if (limitsFromPv && value?.display.controlRange) {
     min = value.display.controlRange?.min;
     max = value.display.controlRange?.max;
   }
-  const numValue = value?.getDoubleValue() || 0;
-  let fraction = 0;
-  if (logScale) {
-    fraction =
-      (Math.log10(numValue) - Math.log10(min)) /
-      (Math.log10(max) - Math.log10(min));
-  } else {
-    fraction = (numValue - min) / (max - min);
-  }
-
-  const onPercent = numValue < min ? 0 : numValue > max ? 100 : fraction * 100;
-  // Store styles in these variables
-  // Change the direction of the gradient depending on wehether the bar is vertical
-  const direction = horizontal ? "to top" : "to left";
-  let fillStyle: CSSProperties = {
-    left: style.borderWidth,
-    bottom: style.borderWidth,
-    backgroundImage: `linear-gradient(${direction}, ${fillColor.toString()} 50%, #ffffff 130%)`
-  };
-  if (horizontal) {
-    fillStyle = {
-      ...fillStyle,
-      height: "100%",
-      top: style.borderWidth,
-      width: `${onPercent}%`
-    };
-  } else {
-    fillStyle = {
-      ...fillStyle,
-      width: "100%",
-      left: style.borderWidth,
-      height: `${onPercent}%`
-    };
-  }
+  const numValue = value?.getDoubleValue() ?? 0;
+  const percent = logScale
+    ? (Math.log10(numValue) - Math.log10(min)) /
+      (Math.log10(max) - Math.log10(min))
+    : (numValue - min) / (max - min);
```

Comment:
I'm getting a different amount of the progress bar shown as filled compared to Phoebs/the old widget. It's worth double checking the calculations of percentage fill or anything else related to fill amount.